### DISCUSSION TITLE
[Enhancement] [cascader] <El-Cascader>是否有考虑为collapse-tags-tooltip增加样式上的控制

### DISCUSSION THREAD

**Discussion by HJ-Hao** [element-plus/element-plus#16798] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `cascader`

## Description

在设置 collapse-tags 和 collapse-tags-tooltip 属性时，发现如果用户有选择了太多的项，el-cascader 内部的 el-tooltip 元素高度有可能会超出屏幕高度。我想编辑 CSS 样式来解决这个问题。但唯一的方法是改变全局类 .el-popper。是否可能添加一个新属性像 popper class 来控制这个类？或者可以有其他解决办法吗？感谢回答

[online demo](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGRpdiBjbGFzcz1cIm0tNFwiPlxuICAgIDxwPkNvbGxhcHNlIHRhZ3MgdG9vbHRpcDwvcD5cbiAgICA8ZWwtY2FzY2FkZXJcbiAgICAgIDpvcHRpb25zPVwib3B0aW9uc1wiXG4gICAgICA6cHJvcHM9XCJwcm9wc1wiXG4gICAgICBjb2xsYXBzZS10YWdzXG4gICAgICBjb2xsYXBzZS10YWdzLXRvb2x0aXBcbiAgICAgIGNsZWFyYWJsZVxuICAgIC8+XG4gIDwvZGl2PlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmNvbnN0IHByb3BzID0geyBtdWx0aXBsZTogdHJ1ZSB9XG5cbmNvbnN0IG9wdGlvbnMgPSBbXG4gIHtcbiAgICB2YWx1ZTogMSxcbiAgICBsYWJlbDogJ0FzaWEnLFxuICAgIGNoaWxkcmVuOiBbXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiAyLFxuICAgICAgICBsYWJlbDogJ0NoaW5hJyxcbiAgICAgICAgY2hpbGRyZW46IFtcbiAgICAgICAgICB7IHZhbHVlOiAzLCBsYWJlbDogJ0JlaWppbmcnIH0sXG4gICAgICAgICAgeyB2YWx1ZTogNCwgbGFiZWw6ICdTaGFuZ2hhaScgfSxcbiAgICAgICAgICB7IHZhbHVlOiA1LCBsYWJlbDogJ0hhbmd6aG91JyB9LFxuICAgICAgICBdLFxuICAgICAgfSxcbiAgICAgIHtcbiAgICAgICAgdmFsdWU6IDYsXG4gICAgICAgIGxhYmVsOiAnSmFwYW4nLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHsgdmFsdWU6IDcsIGxhYmVsOiAnVG9reW8nIH0sXG4gICAgICAgICAgeyB2YWx1ZTogOCwgbGFiZWw6ICdPc2FrYScgfSxcbiAgICAgICAgICB7IHZhbHVlOiA5LCBsYWJlbDogJ0t5b3RvJyB9LFxuICAgICAgICBdLFxuICAgICAgfSxcbiAgICAgIHtcbiAgICAgICAgdmFsdWU6IDEwLFxuICAgICAgICBsYWJlbDogJ0tvcmVhJyxcbiAgICAgICAgY2hpbGRyZW46IFtcbiAgICAgICAgICB7IHZhbHVlOiAxMSwgbGFiZWw6ICdTZW91bCcgfSxcbiAgICAgICAgICB7IHZhbHVlOiAxMiwgbGFiZWw6ICdCdXNhbicgfSxcbiAgICAgICAgICB7IHZhbHVlOiAxMywgbGFiZWw6ICdUYWVndScgfSxcbiAgICAgICAgXSxcbiAgICAgIH0sXG4gICAgXSxcbiAgfSxcbiAge1xuICAgIHZhbHVlOiAxNCxcbiAgICBsYWJlbDogJ0V1cm9wZScsXG4gICAgY2hpbGRyZW46IFtcbiAgICAgIHtcbiAgICAgICAgdmFsdWU6IDE1LFxuICAgICAgICBsYWJlbDogJ0ZyYW5jZScsXG4gICAgICAgIGNoaWxkcmVuOiBbXG4gICAgICAgICAgeyB2YWx1ZTogMTYsIGxhYmVsOiAnUGFyaXMnIH0sXG4gICAgICAgICAgeyB2YWx1ZTogMTcsIGxhYmVsOiAnTWFyc2VpbGxlJyB9LFxuICAgICAgICAgIHsgdmFsdWU6IDE4LCBsYWJlbDogJ0x5b24nIH0sXG4gICAgICAgIF0sXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICB2YWx1ZTogMTksXG4gICAgICAgIGxhYmVsOiAnVUsnLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHsgdmFsdWU6IDIwLCBsYWJlbDogJ0xvbmRvbicgfSxcbiAgICAgICAgICB7IHZhbHVlOiAyMSwgbGFiZWw6ICdCaXJtaW5naGFtJyB9LFxuICAgICAgICAgIHsgdmFsdWU6IDIyLCBsYWJlbDogJ01hbmNoZXN0ZXInIH0sXG4gICAgICAgIF0sXG4gICAgICB9LFxuICAgIF0sXG4gIH0sXG4gIHtcbiAgICB2YWx1ZTogMjMsXG4gICAgbGFiZWw6ICdOb3J0aCBBbWVyaWNhJyxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICB2YWx1ZTogMjQsXG4gICAgICAgIGxhYmVsOiAnVVMnLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHsgdmFsdWU6IDI1LCBsYWJlbDogJ05ldyBZb3JrJyB9LFxuICAgICAgICAgIHsgdmFsdWU6IDI2LCBsYWJlbDogJ0xvcyBBbmdlbGVzJyB9LFxuICAgICAgICAgIHsgdmFsdWU6IDI3LCBsYWJlbDogJ1dhc2hpbmd0b24nIH0sXG4gICAgICAgIF0sXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICB2YWx1ZTogMjgsXG4gICAgICAgIGxhYmVsOiAnQ2FuYWRhJyxcbiAgICAgICAgY2hpbGRyZW46IFtcbiAgICAgICAgICB7IHZhbHVlOiAyOSwgbGFiZWw6ICdUb3JvbnRvJyB9LFxuICAgICAgICAgIHsgdmFsdWU6IDMwLCBsYWJlbDogJ01vbnRyZWFsJyB9LFxuICAgICAgICAgIHsgdmFsdWU6IDMxLCBsYWJlbDogJ090dGF3YScgfSxcbiAgICAgICAgXSxcbiAgICAgIH0sXG4gICAgXSxcbiAgfSxcbiAge1xuICAgIHZhbHVlOiAzMixcbiAgICBsYWJlbDogJ05vcnRoIEFtZXJpY2EnLFxuICAgIGNoaWxkcmVuOiBbXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiAzMyxcbiAgICAgICAgbGFiZWw6ICdVUycsXG4gICAgICAgIGNoaWxkcmVuOiBbXG4gICAgICAgICAgeyB2YWx1ZTogMzQsIGxhYmVsOiAnTmV3IFlvcmsnIH0sXG4gICAgICAgICAgeyB2YWx1ZTogMzUsIGxhYmVsOiAnTG9zIEFuZ2VsZXMnIH0sXG4gICAgICAgICAgeyB2YWx1ZTogMzYsIGxhYmVsOiAnV2FzaGluZ3RvbicgfSxcbiAgICAgICAgXSxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiAzNyxcbiAgICAgICAgbGFiZWw6ICdDYW5hZGEnLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHsgdmFsdWU6IDM4LCBsYWJlbDogJ1Rvcm9udG8nIH0sXG4gICAgICAgICAgeyB2YWx1ZTogMzksIGxhYmVsOiAnTW9udHJlYWwnIH0sXG4gICAgICAgICAgeyB2YWx1ZTogNDAsIGxhYmVsOiAnT3R0YXdhJyB9LFxuICAgICAgICBdLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuICB7XG4gICAgdmFsdWU6IDMyLFxuICAgIGxhYmVsOiAnTm9ydGggQW1lcmljYScsXG4gICAgY2hpbGRyZW46IFtcbiAgICAgIHtcbiAgICAgICAgdmFsdWU6IDMzLFxuICAgICAgICBsYWJlbDogJ1VTJyxcbiAgICAgICAgY2hpbGRyZW46IFtcbiAgICAgICAgICB7IHZhbHVlOiAzNCwgbGFiZWw6ICdOZXcgWW9yaycgfSxcbiAgICAgICAgICB7IHZhbHVlOiAzNSwgbGFiZWw6ICdMb3MgQW5nZWxlcycgfSxcbiAgICAgICAgICB7IHZhbHVlOiAzNiwgbGFiZWw6ICdXYXNoaW5ndG9uJyB9LFxuICAgICAgICBdLFxuICAgICAgfSxcbiAgICAgIHtcbiAgICAgICAgdmFsdWU6IDM3LFxuICAgICAgICBsYWJlbDogJ0NhbmFkYScsXG4gICAgICAgIGNoaWxkcmVuOiBbXG4gICAgICAgICAgeyB2YWx1ZTogMzgsIGxhYmVsOiAnVG9yb250bycgfSxcbiAgICAgICAgICB7IHZhbHVlOiAzOSwgbGFiZWw6ICdNb250cmVhbCcgfSxcbiAgICAgICAgICB7IHZhbHVlOiA0MCwgbGFiZWw6ICdPdHRhd2EnIH0sXG4gICAgICAgIF0sXG4gICAgICB9LFxuICAgIF0sXG4gIH0sXG5dXG48L3NjcmlwdD5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIl9vIjp7fX0=)

### Additional comments

<img width="649" alt="image" src="https://github.com/element-plus/element-plus/assets/50818521/968025eb-9c66-4e34-97dd-be100ce4a4e8">

**Comment by warmthsea** [comment]
你可以通过 popper-class 设置自定义类名
然后设置 max-height 和 overflow auto 让其滚动来解决这个问题

**Reply by HJ-Hao** [reply]
感谢回答，但popper-class的自定义类名只能控制下方选项，而不能控制这个tooltip的样式。
<img width="324" alt="image" src="https://github.com/element-plus/element-plus/assets/50818521/33bee1f5-88aa-47d5-bd1b-3f0046e5ad5e">
<img width="240" alt="image" src="https://github.com/element-plus/element-plus/assets/50818521/4910e7c3-7c31-4d5c-b06b-a4b6781f71a7">

**Comment by warmthsea** [comment]
![screenshots](https://github.com/element-plus/element-plus/assets/45450994/b53543dc-7f1d-42a0-9f78-6873088c0761)

**Reply by HJ-Hao** [reply]
ok清楚了，感谢大佬

**Reply by wuwenyuan111** [reply]
这仅限只有一个cascader组件的时候，如果多个的话，或是有select组件的话，就会不一样了。大佬有什么解决方法吗
